src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -73,14 +70,22 @@ private async Task RegisterSdkRuntimeNeededInProjectAsync(ConfiguredProject proj
 
                 string? TargetFrameworkVersion = await configuration.TargetFrameworkVersion.GetDisplayValueAsync();
 
+                string? componentId = GetComponentId(TargetFrameworkIdentifier, TargetFrameworkVersion);
+                
+                _missingSetupComponentRegistrationService.RegisterSdkRuntimeComponentId(_projectGuid, project, componentId);
+            }
+
+            static string? GetComponentId(string TargetFrameworkIdentifier, string TargetFrameworkVersion)
```

Comment:
2 things:
- Why are the parameters changing from nullable to non-nullable (variables above the `GetComponentId` call are nullable but the parameters are non-nullable)?
- nit: Function parameters should start with a lowercase letter.

Reply:
`RegisterSdkRuntimeNeededInProjectAsync()` used to skip the execution of `RegisterSdkRuntimeComponentId()` when there is no valid runtime. Now we always call `RegisterSdkRuntimeComponentId()` and use null to say that there is no valid runtime is found.